### PR TITLE
Handle CSV input of nullary relations

### PR DESCRIPTION
**Pull request by mmcgr**
Add handling of nullary input from file. This already existed on most platforms, but relied on undefined behaviour.

Nullary input from other forms of IO is explicitly not handled. Current behaviour is to silently fail.

### CODE REVIEW

**Comment by mmcgr** [comment]
Jenkins, test this please

File: src/ReadStreamCSV.h [hunk 1]
```diff
@@ -49,6 +49,22 @@ class ReadStreamCSV : public ReadStream {
     ~ReadStreamCSV() override = default;
 
 protected:
+    /**
+     * Read and return true if there is content..
```
Comment (azreika) [inline src/ReadStreamCSV.h]:
```suggestion
     * Read and return true if there is content.
```

**Comment by azreika** [comment]
Where did it fail previously? On `readNextTuple()`? If so, is it appropriate to put in an assertion after grabbing the result to check that it isn't null so it's easier to debug if it causes problems later on?

**Comment by b-scholz** [comment]
You may need to reformat.

**Comment by mmcgr** [comment]
> Where did it fail previously? On `readNextTuple()`? If so, is it appropriate to put in an assertion after grabbing the result to check that it isn't null so it's easier to debug if it causes problems later on?

We're using the existence of a line, blank or otherwise, to set up a nullary. So in our tests we read in a blank line, line.back() was then called to check for Windows line endings - and line.back() was correctly asserting that back() is not called on an empty string.

Nullaries don't need extra error handling now, but non-nullaries reading from invalid files that have blank lines, and aren't using Windows line endings, may experience similar issues. Ignoring the potential error makes the code marginally faster, but potentially problematic.

I could test if there is a performance difference when reading large files. It's a small cost, but in a method that is called quite frequently. If it's small enough, I could replace this entire PR with a size check and revert to our previous handling of nullaries (hope that no one is using non-csv and nullary input).

**Comment by mmcgr** [comment]
> You may need to reformat.

There is a trial on Jenkins of running clang-tidy with some standard tests. This is returning errors as we have some code that does not follow our standards. Passing the test is not required, but I will add a separate PR to fix the style check so the results are more useful.

**Comment by mmcgr** [comment]
In testing with a 20GB file read from nvme the speeds were indistinguishable between original, this, or adding a size check to readNext. Slower storage mediums would only lessen the difference, if there is one. I'm closing this PR in favour of a much simpler change.

Adding a nullary specific method may still be added in future if we add full nullary support to other input methods

**Pull request closed without merging**